gempy/core/tensor/tensorflow_graph_test.py

```diff
@@ -0,0 +1,647 @@
+            tf.cos(self.deg2rad(azimuth_)) * polarity_
+        G_z = tf.cos(self.deg2rad(dip_angles_)) * polarity_
+
+        G = tf.concat([G_x, G_y, G_z], -1)
+
+        G = tf.expand_dims(G, axis=1)
+        b_vector = tf.pad(G, [[0, length_of_C - G.shape[0]], [0, 0]])
+
+        return b_vector
+
+    
+    def solve_kriging(self, dips_position_all, ref_layer_points, rest_layer_points, b=None):
+
+        C_matrix = self.covariance_matrix(
+            dips_position_all, ref_layer_points, rest_layer_points)
+
+        b_vector = self.b_vector()
+
+
+
+        DK = tf.linalg.solve(C_matrix, b_vector)
+
+        return DK
+
+    
+    def x_to_interpolate(self, grid, ref_layer_points, rest_layer_points):
+        grid_val = tf.concat([grid, rest_layer_points], 0)
+        grid_val = tf.concat([grid_val, ref_layer_points], 0)
+
+        return grid_val
+
+    
+    def extend_dual_kriging(self, weights, grid_shape):
+
+        DK_parameters = weights
+
+        DK_parameters = tf.tile(DK_parameters, [1, grid_shape])
+
+        return DK_parameters
+
+    
+    def contribution_gradient_interface(self, dips_position_all, grid_val=None, weights=None):
+        dips_position_all_tiled = tf.tile(
+            dips_position_all, [self.n_dimensions, 1])
+        length_of_CG = self.matrices_shapes()[0]
+
+        hu_SimPoint = self.cartesian_dist(dips_position_all, grid_val)
+
+        sed_dips_SimPoint = self.squared_euclidean_distance(
+            dips_position_all_tiled, grid_val)
+
+        sigma_0_grad = tf.reduce_sum((weights[:length_of_CG] *
+                                      self.gi_rescale * (tf.negative(hu_SimPoint) *\
+                                                          # first derivative
+                                                          tf.where((sed_dips_SimPoint < self.a_T_surface),
+                                                                   x=(- self.c_o_T * ((-14 / self.a_T_surface ** 2) + 105 / 4 * sed_dips_SimPoint / self.a_T_surface ** 3 -
+                                                                                      35 / 2 * sed_dips_SimPoint ** 3 / self.a_T_surface ** 5 +
+                                                                                      21 / 4 * sed_dips_SimPoint ** 5 / self.a_T_surface ** 7)),
+                                                                   y=tf.constant(0, dtype=self.dtype)))), 0)
+
+
+        
+        return sigma_0_grad
+
+    
+    def contribution_interface(self, ref_layer_points, rest_layer_points, grid_val, weights=None):
+
+        length_of_CG, length_of_CGI = self.matrices_shapes()[:2]
+
+        # Euclidian distances
+        sed_rest_SimPoint = self.squared_euclidean_distance(
+            rest_layer_points, grid_val)
+        sed_ref_SimPoint = self.squared_euclidean_distance(
+            ref_layer_points, grid_val)
+
+        sigma_0_interf = tf.reduce_sum(-weights[length_of_CG:length_of_CG + length_of_CGI, :] *
```

Comment:
I suggest breaking apart this large block of code to manageable portions or into a helper function to reduce text wrapping